OpenRA.Game/Traits/Player/PlayerResources.cs

```diff
@@ -79,8 +79,35 @@ public bool TakeResources(int num)
 
 		public void GiveCash(int num)
 		{
-			Cash += num;
-			Earned += num;
+			if (Cash < int.MaxValue)
```

Comment:
This isn't redundant - it prevents the (otherwise guaranteed) exception throw and catch if the limit has already been reached.  The perf overhead of the if() is completely negligible.